R/default.R

```diff
@@ -273,6 +273,12 @@ gsynth.default <- function(formula = NULL,data, # a data frame (long-form)
         cat("\nFor clustered bootsrap, please use the nonparametric procedure.\n")
     }
 
+    # Do not attempt to use the parametric bootstrap if the number of treated units is fewer than 40
+    n_treated = length(unique(data[data[,D] ==  1, index[1]]))
+    if (inference == "nonparametric" && n_treated < 40 && se) {
+        stop("Nonparametric bootstrap is inappropriate when there are fewer than 40 treated units")
+    }
+
```

Comment:
These lines check that the number of treated units is greater than 40 if the nonparametric bootstrap has been selected. Sorry about all the other edits; my editor automatically removes trailing whitespace.